epinio/epinio-end-to-end-tests
# Cypress/test new manifest actions UI

## Test for manifest actions in UI ([issue #151](https://github.com/epinio/epinio-end-to-end-tests/issues/151))
---

### Test summary:
Test name: "**Download manifest from ui and push an app from it**"
Location: **applications.specs.ts**

Steps:

- Create an app with 2 instances, 1 config, 1 custom route and 1 env vars
- Download manifest
- Delete app
- Create an app uploading solely the downloaded manifest
- Verify values

### Test evidence:
- Tested locally on standalone UI headless and non-headless mode
![image](https://user-images.githubusercontent.com/37271841/170694118-9c2a61cc-36de-4528-b389-1e0ec9ec8fbd.png)

https://user-images.githubusercontent.com/37271841/170693954-97d71a44-01d2-4a43-801e-dbc733bd4e5c.mp4



### Additional notes:
The initial plan for verifying the the values were meant to be done on the details page without creating the app to save time. However there were difficulties locating the displayed values on the shadow DOM. I tried several things and did not manage to point correctly at all elements; for this reason verification of the test was done after pushing the app. Since this adds more time, if at some point is possible to reach these elements, we could eliminate the need of pushing the app and saving some time. 

![image](https://user-images.githubusercontent.com/37271841/170694461-a0eb11de-fdd7-4bb2-b856-48087d3a2655.png)

Related links on shadow elements with Cypress:
- [cypress-shadow-dom](https://www.npmjs.com/package/cypress-shadow-dom)
- https://docs.cypress.io/api/commands/shadow#Known-Issue
- https://sahithigundu.blogspot.com/2020/09/how-to-access-shadow-dom-elements-in.html
